### PR TITLE
fix: move shared dataset-dumping code to speechain/datasets, rename datasets/ to data/

### PR DESCRIPTION
**Pull request by bagustris**
## Summary
- Moves the shared dataset-dumping Python code (`meta_generator.py`, `meta_post_processor.py`, `pyscripts/`) from the top-level `datasets/` folder into `speechain/datasets/`, so it's installed with the package and covered by CI (Black/Ruff, which only scan `speechain/`) — closes #2
- Merges `speechain/dataset/` into `speechain/datasets/` so all dataset-related code lives in one sub-package
- Renames the top-level `datasets/` folder to `data/` to avoid confusion with `speechain.datasets` and the HuggingFace `datasets` package
- Fixes remaining stale `datasets/` path references and broken GitHub links left over from the rename (docstrings, `.gitignore`, `docs/handbook.md`, `recipes/asr/README.md`, `data/README.md`)
- Fixes an unrelated pre-existing bug in `speechain/utilbox/dump_util.py` found while reviewing the migrated pyscripts: `get_readable_number`/`parse_readable_number` crashed on non-round values (float contamination via `%= 1e9`-style literals) and on any multi-unit string like `"5m3k"` or `"1b500m"` (wrong splitter + broken remainder handling)

## Test plan
- [x] All migrated modules under `speechain/datasets/` and `speechain/datasets/pyscripts/` import cleanly and compile
- [x] All migrated pyscripts run `--help` successfully as standalone scripts (matching how `data/data_dumping.sh` invokes them)
- [x] Verified previously-claimed bug fixes in the branch (`.lab` file handling, `np.loadtxt` `ndmin=2`, punctuation-start `IndexError`, `txt_format` default) are correctly implemented
- [x] Manually ran `tests/utilbox/test_dump_util.py` assertions (pytest unavailable in this sandbox), including new round-trip tests for billion-level and multi-unit values that previously crashed
- [ ] Run the full `pytest tests/` and `black`/`ruff` checks in a full dev environment

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_017dYBm8dAAhSHCR56SgD2UC

## Summary by Sourcery

Move dataset code into the speechain.datasets package, rename the top-level datasets/ directory to data/, and fix related documentation, scripts, and utilities.

Bug Fixes:
- Correct get_readable_number/parse_readable_number in dump_util to handle large and multi-unit values robustly and align tests with the new behavior.
- Fix regular expressions for punctuation cleanup in en_text_process by using raw string patterns.
- Ensure dataset-related helper scripts handle edge cases such as single-line metadata files and existing .lab files without aborting processing.
- Update tests and utilities to reflect the new data directory structure and to gracefully skip optional dependencies when unavailable.

Enhancements:
- Consolidate dataset-related Python modules under the new speechain.datasets package, including moving shared dumping scripts (pyscripts) into the package so they are installed and covered by CI.
- Rename the top-level datasets/ directory to data/ across the project to avoid name clashes and clarify its role as a data root, updating references in docs, configs, and scripts accordingly.
- Adjust iterator and test imports to use speechain.datasets instead of speechain.dataset, and update package metadata so all speechain sub-packages are installed without exposing a top-level datasets package.
- Document the new data and speechain/datasets layout in docs, READMEs, and contributor guidance.

Build:
- Change setup.py to install all speechain sub-packages while removing the old top-level datasets package to avoid conflicts with third-party libraries.

Documentation:
- Rewrite dataset-related docs and examples to reference the new data/ folder, updated script locations, and the speechain.datasets package.
- Update MkDocs navigation and internal links to point to data.md and the merged speechain/datasets documentation.

Tests:
- Relocate dataset tests to tests/datasets, update imports to the new package layout, and add round-trip tests for readable-number parsing, including billion-level and multi-unit values.

### CODE REVIEW

File: speechain/utilbox/dump_util.py [hunk 1]
```diff
@@ -168,21 +168,21 @@ def get_readable_number(raw_number: int or float) -> str:
...
-        raw_number %= 1e6
+    if raw_number // 1_000_000 > 0:
+        read_number += f"{raw_number // 1_000_000}m"
+        raw_number %= 1_000_000
     # kilo-level
-    if raw_number // 1e3 > 0:
-        read_number += f"{int(raw_number // 1e3)}k"
-        raw_number %= 1e3
+    if raw_number // 1_000 > 0:
+        read_number += f"{raw_number // 1_000}k"
```
[the remark this answers is not in the document]
Comment (bagustris) [reply speechain/utilbox/dump_util.py:180]: False positive. get_readable_number casts raw_number to int at line 164-167 before any ///%=, so it stays int throughout. Verified: get_readable_number(1_500_000_000.0) → '1b500m', no crash.
